src/main/java/org/jruby/ext/openssl/Cipher.java

```diff
@@ -1104,6 +1104,11 @@ private String getCipherAlgorithm() {
 
     @JRubyMethod
     public IRubyObject update(final ThreadContext context, final IRubyObject arg) {
+        return update(context, arg, null);
+    }
+
+    @JRubyMethod
+    public IRubyObject update(final ThreadContext context, final IRubyObject arg, IRubyObject buffer) {
```

Comment:
As a @JRubyMethod buffer can be any Ruby type.  So you need to do something like TypeConverter.checkStringType.

Reply:
thanks